### ISSUE TITLE
zeitwerk:check fails on 6.0.2

### ISSUE THREAD

**Issue by shouichi** [rails/rails#37966]
### Steps to reproduce

`zeitwerk:check` fails on a newly created project.

```
$ rails new foo
$ cd foo
$ ./bin/rails zeitwerk:check
```

This might be related to #37756.

### Expected behavior

Just works.

### Actual behavior

```
Hold on, I am eager loading the application.
rails aborted!
NoMethodError: undefined method `eager_load_paths' for #<I18n::Config:0x000055fe6c0abef8>
/usr/local/bundle/gems/railties-6.0.2/lib/rails/tasks/zeitwerk.rake:57:in `block (3 levels) in <main>'
/usr/local/bundle/gems/railties-6.0.2/lib/rails/tasks/zeitwerk.rake:56:in `map'
/usr/local/bundle/gems/railties-6.0.2/lib/rails/tasks/zeitwerk.rake:56:in `block (2 levels) in <main>'
/usr/local/bundle/gems/railties-6.0.2/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/usr/local/bundle/gems/railties-6.0.2/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/usr/local/bundle/gems/railties-6.0.2/lib/rails/command.rb:48:in `invoke'
/usr/local/bundle/gems/railties-6.0.2/lib/rails/commands.rb:18:in `<main>'
/usr/local/bundle/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
/usr/local/bundle/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
/usr/local/bundle/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/usr/local/bundle/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
/usr/local/bundle/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/usr/local/bundle/gems/activesupport-6.0.2/lib/active_support/dependencies.rb:325:in `block in require'
/usr/local/bundle/gems/activesupport-6.0.2/lib/active_support/dependencies.rb:291:in `load_dependency'
/usr/local/bundle/gems/activesupport-6.0.2/lib/active_support/dependencies.rb:325:in `require'
./bin/rails:4:in `<main>'
Tasks: TOP => zeitwerk:check
(See full trace by running task with --trace)
```

### System configuration
**Rails version**: 6.0.2

**Ruby version**: ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-linux]

**Comment by benoitongit** [comment]
Seeing the exact same issue

**Comment by brian-kephart** [comment]
Same. Downgrading to 6.0.1 works with the same version of zeitwerk (2.2.2).

**Comment by kaspth** [comment]
Cc @fxn

**Comment by nsommer** [comment]
Same, we have the zeitwerk check in our CI scripts and on Rails 6.0.2 it fails for the same reason.

**Comment by fxn** [comment]
I’ll have a look soon.

**Comment by single-stop-nicholalexander** [comment]
Not sure if it is any help but was looking at this just now.  This looks like an issue with the config object in i18n.  The zeitwerk rake task is asking each of the config objects for their `eager_load_paths`.  The i18n config object is the only one in the usual list of the Rails.configuration.eager_load_namespaces that doesn't define `eager_load_paths`.  

I'm entirely not sure how to investigate further in such as way as to be of any help.

**Comment by fxn** [comment]
I think `i18n` is fine. Objects in `eager_load_namespaces` [are only required to respond to `eager_load!`](https://guides.rubyonrails.org/configuring.html).

That task only cares about files managed by Zeitwerk in the application and engines, it should support the `i18n` case. I'll write a fix.

**Comment by fxn** [comment]
By the way, for me, this task was a quick & dirty script to help people migrate from `classic` to `zeitwerk` modes. Once in `zeitwerk` mode, I thought its job was done.

It is very cool to know if has use cases beyond the initial intention like running it in CI. Will make it more testable and robust.

**Comment by fxn** [comment]
Fixed in da28edf9baf0bd83b5308564bdcd8f4673b492bd. I'll rewrite this with more time, but meanwhile the regresssion is no longer present in `master` or `6-0-stable`.

**Comment by single-stop-nicholalexander** [comment]
amazing, thanks!

**Comment by brian-kephart** [comment]
Thank you @fxn! FWIW, I find it useful to run this command when installing a new dependency.

**Comment by shouichi** [comment]
Thanks for the fix!

We run the `zeitwerk:check` task on CI to check if the file/class names are correct.

**Comment by rodinux** [comment]
I have always the same issue using rvm stable branch. 
```
$ rails zeitwerk:check --trace
** Invoke zeitwerk:check (first_time)
** Invoke environment (first_time)
** Execute environment
warning package.json: No license field                                         
warning No license field                                                       
** Execute zeitwerk:check
Hold on, I am eager loading the application.
rails aborted!
NoMethodError: undefined method `eager_load_paths' for #<I18n::Config:0x00005603db089fc8>
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/tasks/zeitwerk.rake:57:in `block (3 levels) in <main>'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/tasks/zeitwerk.rake:56:in `map'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/tasks/zeitwerk.rake:56:in `block (2 levels) in <main>'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/task.rb:281:in `block in execute'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/task.rb:281:in `each'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/task.rb:281:in `execute'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/home/rodebian/.rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0/monitor.rb:230:in `mon_synchronize'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/task.rb:199:in `invoke_with_call_chain'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/task.rb:188:in `invoke'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:160:in `invoke_task'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:116:in `each'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:116:in `block in top_level'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:125:in `run_with_threads'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:110:in `top_level'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/lib/rake/application.rb:186:in `standard_exception_handling'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/command.rb:48:in `invoke'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/railties-6.0.2.1/lib/rails/commands.rb:18:in `<main>'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.2.1/lib/active_support/dependencies.rb:325:in `block in require'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.2.1/lib/active_support/dependencies.rb:291:in `load_dependency'
/home/rodebian/.rvm/gems/ruby-2.6.3/gems/activesupport-6.0.2.1/lib/active_support/dependencies.rb:325:in `require'
bin/rails:4:in `<main>'
Tasks: TOP => zeitwerk:check
```
# System configuration
## Rails version: 
Rails 6.0.2.1
## Ruby version: 
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-linux]

**Comment by pedrofurtado** [comment]
@fxn This bugfix will be available in rails 6.0.3? 🤝

**Comment by fxn** [comment]
@pedrofurtado yes!

**Comment by jrochkind** [comment]
Any good workaround in the meantime to be able to run `zeitwerk:check` on Rails 6.0.2(.1)?  Annoying not to have this tool available.

**Comment by LeFnord** [comment]
@fxn … first thanks for your good work 

when will the fix be released?

**Comment by fxn** [comment]
Thanks!

Let me ask @rafaelfranca... are there any plans for 6.0.3?

**Comment by enowmbi** [comment]
Having the same issue here -- using Ruby 2.7.0 and Rails 6.0.1.2  on Ubuntu 18.04.

**Comment by cesarjr** [comment]
Having same issue here - using ruby 2.6.5p114 and Rails 6.0.2.1 on OSX.

**Comment by say8425** [comment]
Having same issue here - using ruby 2.6.6p114 and Rails 6.0.2.2 on macOS

**Comment by shouichi** [comment]
It seems like the issue is now solved in 6.0.3. Thanks for the work!
